cli/src/main/kotlin/com/bazel_diff/bazel/BazelQueryService.kt

```diff
@@ -35,41 +73,65 @@ class BazelQueryService(
                 add("--bazelrc=/dev/null")
             }
             addAll(startupOptions)
-            add("query")
+            if (useCquery) {
+                add("cquery")
+                add("--transitions=lite")
+            } else {
+                add("query")
+            }
             add("--output")
-            add("streamed_proto")
-            add("--order_output=no")
-            if (keepGoing != null && keepGoing) {
+            if (useCquery) {
+                if (outputCompatibleTargets) {
+                    add("starlark")
+                    add("--starlark:file")
+                    val cqueryOutputFile = Files.createTempFile(null, ".cquery").toFile()
+                    cqueryOutputFile.deleteOnExit()
+                    cqueryOutputFile.writeText("""
+                    def format(target):
+                        if "IncompatiblePlatformProvider" not in providers(target):
+                            label = str(target.label)
+                            if label.startswith("@//"):
+                                # normalize label to be consistent with content inside proto
+                                return label[1:]
+                            else:
+                                return label
+                        return ""
+                    """.trimIndent())
+                    add(cqueryOutputFile.toString())
+                } else {
+                    // Unfortunately, cquery does not support streamed_proto yet.
+                    // See https://github.com/bazelbuild/bazel/issues/17743. This poses an issue for large monorepos.
+                    add("proto")
```

Comment:
Bummer here but seems unavoidable for now

Reply:
Yeah :(. BTW, I realized there is a bug with the first commit so I have pushed another update with one more test covering the basic source code change case.

It seems that the implementation can be cleaner if source code hashing and rule target hashing are done uniformly with a `deps` query to include everything (rather than separate queries on both). But that change would be pretty big so I didn't attempt it.